### ISSUE TITLE
go vet -mod=vendor break if GOFLAGS env sets mod too

### ISSUE THREAD

**Issue by tnozicka** [golang/go#35955]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/tnozicka/.cache/go-build"
GOENV="/home/tnozicka/.config/go/env"
GOEXE=""
GOFLAGS="-mod=vendor"
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/tnozicka/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/tnozicka/lib/go1.13"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/tnozicka/lib/go1.13/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build193936079=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

### What did you do?
```
$ env | grep GOFLAGS
GOFLAGS=-mod=vendor

$ ~/lib/go1.13/bin/go vet -mod=vendor ./...
go vet: mod flag may be set only once
run "go help vet" for more information
```

### What did you expect to see?
No error, since the flag is specified only once on the command line. Other commands work fine in this configurations.

As a library author providing the build system I want to enforce vendoring mode using the `-mod=vendor` and I can't influence what user has in set in his env. The general rule is also that flags win over env, but in this case I think it is really just about handling it gracefully in the implementation internals.

### What did you see instead?
`go vet: mod flag may be set only once`

**Comment by mvdan** [comment]
I'm pretty sure this is a duplicate of https://github.com/golang/go/issues/32471, but cc-ing @bcmills just in case.

**Comment by bcmills** [comment]
Agreed.

**Comment by bcmills** [comment]
Duplicate of #32471
